cmd/consumer/main.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"errors"
+	"net/http"
+	"os"
+	"runtime/debug"
+
+	"github.com/ONSdigital/dp-import-reporter/config"
+
+	"github.com/ONSdigital/dp-import-reporter/handler"
+	"github.com/ONSdigital/dp-import-reporter/schema"
+	"github.com/ONSdigital/go-ns/kafka"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/coocood/freecache"
+)
+
+// var running bool
+
+// logFatal is a utility method for a common failure pattern in main()
+func logFatal(context string, err error, data log.Data) {
+	log.ErrorC(context, err, data)
+	panic(err)
+}
+
+func main() {
+
+	cfg, err := config.Get()
+	if err != nil {
+		log.Error(err, nil)
+		os.Exit(1)
+	}
+
+	newInstanceEventConsumer, err := consumerInit(cfg)
+	if err != nil {
+		logFatal("error initiating", err, nil)
+	}
+	//cache init
+	c := cacheSetup()
+	client := &http.Client{}
+
+	loop(newInstanceEventConsumer, cfg, client, c)
+}
+
+func consumerInit(cfg *config.Config) (*kafka.ConsumerGroup, error) {
+	log.Namespace = "dp-event-reporter"
+
+	log.Info("starting", log.Data{
+		"new-import-topic": cfg.NewInstanceTopic,
+		"import-api":       cfg.ImportAPIURL,
+	})
+	newInstanceEventConsumer, err := kafka.NewConsumerGroup(cfg.Brokers, cfg.NewInstanceTopic, log.Namespace, kafka.OffsetNewest)
+	if err != nil {
+		logFatal("could not obtain consumer", err, nil)
+	}
+	return newInstanceEventConsumer, err
+}
+
+func cacheSetup() *freecache.Cache {
+	cacheSize := 100 * 1024 * 1024
+	c := freecache.NewCache(cacheSize)
+	debug.SetGCPercent(20)
+	return c
+}
+
+func loop(newInstanceEventConsumer *kafka.ConsumerGroup, cfg *config.Config, client *http.Client, c *freecache.Cache) {
```

Comment:
Sorry to be picky but I really don't like the name loop - it doesn't explain what it does - maybe consume(..) instead?